addons/l10n_it_edi/tools/remove_signature.py

```diff
@@ -36,7 +36,11 @@ def remove_signature(content):
     # This method is deprecated, but there are actually no alternatives
     with warnings.catch_warnings():
         warnings.filterwarnings("ignore", category=DeprecationWarning)
-        loaded_data = ssl_crypto.load_pkcs7_data(ssl_crypto.FILETYPE_ASN1, content)
+        try:
+            loaded_data = ssl_crypto.load_pkcs7_data(ssl_crypto.FILETYPE_ASN1, content)
+        except ssl_crypto.Error:
+            _logger.warning("Error reading the content, PKCS#7 signature missing or invalid. Content will be tentatively used as it is.")
+            return content
```

Comment:
So you mean it is maybe already decoded?

Should we really check for a bad extension applied to a file?

Reply:
Sure, this function may receive files not onlly from user upload but also from the Tax Agency webservices. As they just send the vendor bill one time, and the IAP sends the Odoo instance the file one time, if we don't import it just because of the extension, we've not only an unused vendor bill in the IAP that it's very hard to recover, and it won't appear in Odoo, but we won't also be noticed of that, if not in the logs.
So yes, better to check for a wrong filename, at the moment.